bahmnicore-omod/src/main/resources/liquibase.xml

```diff
@@ -4571,4 +4571,19 @@
         </sql>
     </changeSet>
 
+    <changeSet id="bahmni-core-20231206-bah-3336" author="BAHMNI">
+        <preConditions onFail="MARK_RAN">
+            <sqlCheck expectedResult="1">
+                SELECT COUNT(*) FROM program_attribute_type where name = 'Stage' AND datatype =
+                'org.bahmni.module.bahmnicore.customdatatype.datatype.CodedConceptDatatype';
+            </sqlCheck>
+        </preConditions>
+        <comment>Update datatype of patient stage program attribute type</comment>
+        <sql>
+            UPDATE program_attribute_type
+            SET datatype = 'org.openmrs.customdatatype.datatype.ConceptDatatype'
+            WHERE name = 'Stage';
```

Comment:
AND Datatype = 'org.bahmni.module.bahmnicore.customdatatype.datatype.CodedConceptDatatype' ?

Reply:
Added the datatype condition